tests/unit/amo/reducers/test_home.js

```diff
@@ -74,6 +118,21 @@ describe(__filename, () => {
       expect(homeState.collections[0]).toEqual(null);
     });
 
+    it('returns null for an empty collection', () => {
+      const { store } = dispatchClientMetadata();
+
+      store.dispatch(
+        loadHomeAddons({
+          collections: [{}],
```

Comment:
This should be a valid CollectionAddonsListResponse with an empty set of add-ons, e.g.,

```js
          collections: [
            createFakeCollectionAddonsListResponse({
              addons: [],
            }),
          ],
```